Use `input` and `local` keywords instead of `let` in declarations (#488)
unicode-org/message-format-wg: spec/syntax.md

### Declarations

A **_<dfn>declaration</dfn>_** binds a _variable_ identifier to a value within the scope of a _message_.
This _variable_ can then be used in other _expressions_ within the same _message_.
_Declarations_ are optional: many messages will not contain any _declarations_.

> An **_<dfn>input-declaration</dfn>_** binds a _variable_ to an external input value,
> such as a named argument of a formatting call.

Author: ```suggestion
An **_<dfn>input-declaration</dfn>_** binds a _variable_ to an external input value.
```

> such as a named value in an object argument of a formatting call.

Reviewer: What does this mean? What is _an object argument_?

Reviewer 2: That's a great question. I think we'd be better off defining what input values are and then referring to that. Perhaps:

```suggestion
An **_<dfn>input value</dfn>_** (or _external input value_) is a named value passed to the message formatter
as part of a formatting call. 
> For example, the _placeholder_ `{$user}` in the _message_ `Hello, {$user}` refers to the
> _input value_ named `user` in the input to MessageFormat.

An **_<dfn>input declaration</dfn>_** binds a _variable_ to an _input value_, including an optional
function or other options.
```

Author: I think introducing _input value_ as a defined term would be a bit heavy here. Rather, let me simplify the suggestion a bit. Named arguments aren't directly supported in most languages (they are in at least PHP and Python), but conceptually they should be pretty understandable, and implementable via an explicitly created map argument in ~all languages.

```suggestion
such as a named argument of a formatting call.
```

annotation = (function *(s option)) / reserved / private-use
```

An **_<dfn>operand</dfn>_** is the _literal_ of a _literal-expression_ or

> the _variable_ of a \_variable-expression.

Reviewer: ```suggestion
the _variable_ of a _variable-expression_.
```
